src/OneOff.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace ipl\Scheduler;
+
+use DateTime;
+use ipl\Scheduler\Contract\Frequency;
+
+class OneOff implements Frequency
+{
+    /** @var DateTime Start time of this frequency */
+    protected $dateTime;
+
+    public function __construct(DateTime $dateTime)
+    {
+        $this->dateTime = clone $dateTime;
+    }
+
+    public function isDue(DateTime $dateTime): bool
+    {
+        return ! $this->isExpired($dateTime) && $dateTime == $this->dateTime;
```

Comment:
Should be `$dateTime >= $this->dateTime`. I think it makes sense to additionally check if the frequency is also due for something like time plus 1 hour. Should then also be done for the `Cron` class.

In `isExpired()` and `isDue()` it would also be useful to have recurring `if` patterns of the same style. Either we always put the parameter first or the property, ie.

`$this->start <= $due` vs `$due > $this->start`.

Reply:
> Should be `$dateTime >= $this->dateTime`. I think it makes sense to additionally check if the frequency is also due for something like time plus 1 hour. Should then also be done for the `Cron` class.

`isExpired()` returns `true` if `$dateTime` is greater than `$this->dateTime`, so I can additionally check here if it isn't the same as (**doesn't compare strictly**) `$this->dateTime`. When it isn't, `isDue()` always returns `false` because the frequency has already expired.